docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,107 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, annotations) to individual log entries, and sending the logs to the Elasticsearch cluster. The additional metadata, such as labels, augment log entries to allow users to intelligently query their deployed services.  The metadata acts to enrich the logs so users may logically group their services as a single, coherent application.
+
+The pod metadata is currently sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]; Each watch maintains an open connection to the API server. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  The current deployment topology at scale:
+
+* Results in logs not being shipped to the aggregator and thus being unavailable
+* Limits the number of cluster nodes because of the cap on the number of open connections
+
+This proposal addresses both the short and long term solutions to resolve the connection issues being experienced using the v3.4 EFK stack.
+### Considerations
+1. OpenShift Logging is not the only consumer of OpenShift log messages.  There are customers who desire to use their existing logging infrastructure(e.g. Splunk, Elasticsearch).  Log entries need to be enriched with metadata irrespective of the final destination.
+2. It is speculated that there is a low probability that labels and annotations will change during the lifetime of a running pod. We should determine if 'stale' labels and annotations are acceptable
```

Comment:
This would implies we could  have stale labels now since the server terminates watches after a known amount of time.  It doesn't answer the staleness question but it is something we should fix ASAP